app/plugins/rwahsNavigation/rwahsNavigationPlugin.php

```diff
@@ -0,0 +1,194 @@
+        if (is_array($va_shortcuts)) {
+            foreach ($va_shortcuts as $vs_key => $vs_type_code) {
+                if (!is_string($vs_type_code)) {
+                    array_push($va_errors, _t('Custom new menu shortcut with key "%1" is not a type code (string value)', $vs_key));
+                }
+            }
+        }
+        return $va_errors;
+    }
+
+    private function _checkSearchMenuShortcuts() {
+        $va_errors = array();
+        $va_shortcuts = $this->opo_config->get('search_menu_shortcuts');
+        if (is_array($va_shortcuts)) {
+            foreach ($va_shortcuts as $vs_key => $va_shortcut) {
+                if (!is_array($va_shortcut)) {
+                    array_push($va_errors, _t('Custom search shortcut with key "%1" is not an array', $vs_key));
+                } else {
+                    if (!isset($va_shortcut['type_code']) || empty($va_shortcut['type_code'])) {
+                        array_push($va_errors, _t('Custom search shortcut with key "%1" does not specify a "%2" value, which is required', $vs_key, 'type_code'));
+                    }
+                    if (!isset($va_shortcut['form_code']) || empty($va_shortcut['form_code'])) {
+                        array_push($va_errors, _t('Custom search shortcut with key "%1" does not specify a "%2" value, which is required', $vs_key, 'form_code'));
+                    }
+                    if (!isset($va_shortcut['display_code']) || empty($va_shortcut['display_code'])) {
+                        array_push($va_errors, _t('Custom search shortcut with key "%1" does not specify a "%2" value, which is required', $vs_key, 'display_code'));
+                    }
+                }
+            }
+        }
+        return $va_errors;
+    }
+
+    static function getRoleActionList() {
+        return array();
+    }
+
+    public function hookRenderMenuBar($pa_nav_info) {
+        if (is_array($pa_nav_info['New']['navigation'])) {
+            $pa_nav_info['New']['navigation'] = $this->_getCustomNewMenuItems() + $pa_nav_info['New']['navigation'];
+            unset($pa_nav_info['New']['navigation']['objects']);
+        }
+        if (is_array($pa_nav_info['find']['navigation'])) {
+            $pa_nav_info['find']['navigation'] = $this->_getCustomSearchMenuItems() + $pa_nav_info['find']['navigation'];
+            unset($pa_nav_info['find']['navigation']['objects']);
+        }
+        return $pa_nav_info;
+    }
+
+    private function _getSpacer() {
+        return array(
+            'displayName' => '<div class="sf-spacer"></div>'
+        );
+    }
+
+    private function _getCustomNewMenuItems() {
+        $va_custom_items = array();
+        $va_shortcuts = $this->opo_config->get('new_menu_shortcuts');
+        if (is_array($va_shortcuts)) {
+            foreach ($va_shortcuts as $vs_key => $vs_type_code) {
+                $vo_type = ca_list_items::find(
+                    array( 'idno' => $vs_type_code ),
```

Comment:
Also add a restriction for `list_id` or `list_code` - `idno`'s are only meant to be unique within a specific list.
``` php
$t_subject = $this->opo_datamodel->getInstanceByTableName('ca_objects', true);
$t_list = new ca_lists();
$t_list->load(array('list_code' => $t_subject->getTypeListCode()));
$vn_list_id = $t_list->getPrimaryKey();
```

 Additionally we're probably going to have type based access control, so we'll probably need to check whether users can create objects of each type - as per https://github.com/rwahs/providence/blob/e2df4df82563e5412ff847600dddd0182db809d7/app/lib/ca/BaseEditorController.php#L1183-L1186
and later in that file.